src/ILLink.RoslynAnalyzer/DiagnosticDescriptors.cs

```diff
@@ -59,7 +59,10 @@ public static DiagnosticDescriptor GetDiagnosticDescriptor (DiagnosticId diagnos
 		static string GetDiagnosticCategory (DiagnosticId diagnosticId)
 		{
 			switch ((int) diagnosticId) {
-			case > 2000 and < 3000:
+			case >= 1000 and < 2000:
+				return DiagnosticCategory.TrimmingError;
```

Comment:
Is the category mandatory? In the linker these just get `MessageSubCategory.None`. `TrimmingError` sounds a bit off to me because it suggests that there was an error in the trimmer. If we really need to introduce a category, I think something closer to your original change, maybe `TrimXml` would be more appropriate - but it should only be applied to the specific warning codes that are about Xml.

Reply:
I guess the category isn't mandatory, I just wanted to avoid throwing an exception so I added the category for 1XXX as an error during trimming. Maybe just `Error` makes more sense? Otherwise I could use None and make the `DiagnosticCategory` enum value an empty string.